x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -22,12 +25,21 @@ interface ConstructorOptions {
   isWriteEnabled: boolean;
   waitUntilReadyForReading: Promise<WaitResult>;
   waitUntilReadyForWriting: Promise<WaitResult>;
+  logger: Logger;
 }
 
 export type WaitResult = Either<Error, ElasticsearchClient>;
 
 export class RuleDataClient implements IRuleDataClient {
-  constructor(private readonly options: ConstructorOptions) {}
+  private _isWriteEnabled: boolean = false;
+
+  // Writers cached by namespace
+  private writerCache: Map<string, IRuleDataWriter>;
```

Comment:
Since it's private shall we use `_writerCache` instead?

Reply:
Possibly, I did it with `_isWriteEnabled` as I still wanted the public `isWriteEnabled()` method. Unfortunately getters and setters can't disagree on visibility, so I can't make the `.writeEnabled` getter public, whilst keeping the setter private. Ideally I think it's nicer to just have `public` / `private` denote the visibility.

Reply:
Possibly, I did it with `_isWriteEnabled` as I still wanted the public `isWriteEnabled()` method. Unfortunately getters and setters can't disagree on visibility, so I can't make the `.writeEnabled` getter public, whilst keeping the setter private. Ideally I think it's nicer to just have `public` / `private` denote the visibility.

Reply:
I was confused with the use of `_isWriteEnabled`, but it makes sense now.